dmsrc/iconforge.dm

```diff
@@ -0,0 +1,33 @@
+/// Generates a spritesheet at: [file_path][spritesheet_name]_[size_id].png
+/// Spritesheet will contain all sprites listed within "sprites".
+/// "sprites" format:
+/// list(
+///     "sprite_name" = list(
+///         icon_file = 'icons/path_to/an_icon.dmi',
+///         icon_state = "some_icon_state",
+///         dir = SOUTH,
+///         frame = 1,
+///         transform = list([TRANSFORM_OBJECT], ...)
+///     ),
+///     ...,
+/// )
+/// TRANSFORM_OBJECT format:
+/// list("type" = "BlendColor", "color" = "#ff0000", "blend_mode" = ICON_MULTIPLY)
+/// list("type" = "BlendIcon", "icon" = sprite_object, "blend_mode" = ICON_OVERLAY)
+/// list("type" = "Scale", "width" = 32, "height" = 32)
+/// list("type" = "Crop", "x1" = 0, "y1" = 0, "x2" = 32, "y2" = 32)
+/// Returns a SpritesheetResult as JSON, containing fields:
+/// list(
+///     "sizes" = list("32x32", "64x64", ...etc)
+///     "sprites" = list("sprite_name" = list("size_id" = "32x32", "position" = 0), ...)
+///     "error" = "[A string, empty if there were no errors.]"
+/// )
+/// In the event of an unrecoverable error, where the spritesheet could not even generate, returns a string containing the error.
```

Comment:
This line is vague, please be more clear about how it relates to the above code. On first read I thought this meant
`return "couldn't read shit"` essentially